spec/v2.spec.ts

```diff
@@ -348,6 +350,159 @@ describe('v2', () => {
+          const cloudFnWrap = wrapV2(cloudFn);
+          const cloudEvent = cloudFnWrap().cloudEvent;
+          expect(cloudEvent).deep.equal({
+            // id and time are generated at runtime
+            id: cloudEvent.id,
+            time: cloudEvent.time,
+            specversion: '1.0',
+
+            data: {
+              _data: {
+                ['@type']:
+                  'type.googleapis.com/google.events.firebase.database.v1.ReferenceEventData',
+                data: {},
+                delta: {},
+              },
+              _path: '/foo/bar',
+              app: [undefined],
+              instance: 'instance-1',
+            },
+            instance: 'instance-1',
+            firebaseDatabaseHost: 'firebaseDatabaseHost',
+            ref: '/foo/bar',
+            location: 'us-central1',
+            params: {},
+            source: '',
+            type: 'google.firebase.database.ref.v1.created',
+          });
+        });
+      });
+      describe('database.onRefDeleted()', () => {
+        it('should update CloudEvent appropriately', () => {
+          const referenceOptions = {
+            ref: '/foo/bar',
+            instance: 'instance-1',
+          };
+          const cloudFn = database.onRefDeleted(referenceOptions, handler);
+          const cloudFnWrap = wrapV2(cloudFn);
+          const cloudEvent = cloudFnWrap().cloudEvent;
+          expect(cloudEvent).deep.equal({
+            // id and time are generated at runtime
+            id: cloudEvent.id,
+            time: cloudEvent.time,
+            specversion: '1.0',
+
+            data: {
+              _data: {
+                ['@type']:
+                  'type.googleapis.com/google.events.firebase.database.v1.ReferenceEventData',
+                data: {},
+                delta: {},
+              },
+              _path: '/foo/bar',
+              app: [undefined],
+              instance: 'instance-1',
+            },
+            instance: 'instance-1',
+            firebaseDatabaseHost: 'firebaseDatabaseHost',
+            ref: '/foo/bar',
+            location: 'us-central1',
+            params: {},
+            source: '',
+            type: 'google.firebase.database.ref.v1.deleted',
+          });
+        });
+      });
+      describe('database.onRefUpdated()', () => {
+        it('should update CloudEvent appropriately', () => {
+          const referenceOptions = {
+            ref: '/foo/bar',
+            instance: 'instance-1',
+          };
+          const cloudFn = database.onRefUpdated(referenceOptions, handler);
+          const cloudFnWrap = wrapV2(cloudFn);
+          const cloudEvent = cloudFnWrap().cloudEvent;
+          expect(cloudEvent).deep.equal({
+            // id and time are generated at runtime
+            id: cloudEvent.id,
+            time: cloudEvent.time,
+            specversion: '1.0',
+
+            data: {
+              after: {
+                _data: {},
```

Comment:
Above we had data._data.data and here we have data.after._data. It seems like something is wrong here since the payload should always be the same.

Reply:
There was already a makeDataSnapshotExample from v1 rtdb which I am making use of now.